guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md

```diff
@@ -0,0 +1,355 @@
+---
+layout: single
+title: "ExtensionPoint Plugin Walkthrough"
+sidebar:
+  nav: guides
+---
+
+{% include toc %}
+
+<div class="notice--danger">
+  <strong>Note:</strong> Plugins are an alpha feature that is under active development and may change.
+</div>
+
+
+# Prerequisites
+
+This walkthrough assumes you have read the [Plugin Creators Guide Overview](/guides/developer/plugin-creators-guide/overview/).
+
+
+# pf4jStagePlugin ExtensionPoint plugin
+
+The [pf4jStagePlugin](https://github.com/spinnaker-plugin-examples/pf4jStagePlugin) creates a custom pipeline stage that waits a specified number of seconds before signaling success.
+
+## Structure
+
+The plugin has frontend (`random-wait-deck`) and backend (`random-wait-orca`) components.
+
+The backend consists of five [Kotlin](https://kotlinlang.org/docs/reference/) classes in the `io.armory.plugin.state.wait.random` package:
+
+* `Context.kt`: a data class that stores the `maxWaitTime` value
+* `Output.kt`: a data class that stores the `timeToWait` getValue
+The plugin backend implements the Orca `com.netflix.spinnaker.orca.api.SimpleStage` interface.
+
+# Project Configuration
+
+Keep the following recommendations and requirements in mind:
+- We recommend making the project a Gradle project. There is Gradle tooling to support plugin development.
+- You can add any dependencies you need to make your plugin successful.. *TODO-CF* this is almost certainly a lie when
+  it comes to things that conflict with the dependencies in the compileOnly scope...
+
+## Gradle configuration
+
+Some important aspects of your project's gradle build:
+
+### gradle.properties
+This is optional, but externalizes the versions of tooling and API dependencies from your
+build script.
```

Comment:
This wording is deceptive. Organizing these values into `gradle.properties` is optional, but these versions are very much mandatory.

Reply:
Cameron Fieber wrote the project configuration stuff. I changed the wording to what you suggest.